File: src/main/java/org/cobbzilla/util/network/NetworkPort.java
package org.cobbzilla.util.network;

import lombok.experimental.Accessors;

@Accessors(chain=true)
public class NetworkPort {

    Integer port;
    TransportProtocol protocol = TransportProtocol.tcp;
    NetworkInterfaceType iface = NetworkInterfaceType.world;

    public Integer getPort() {
        return this.port;
    }

    public TransportProtocol getProtocol() {
        return this.protocol;
    }

    public NetworkInterfaceType getIface() {
        return this.iface;
    }

    public NetworkPort setPort(Integer port) {
        this.port = port;
        return this;
    }

    public NetworkPort setProtocol(TransportProtocol protocol) {
        this.protocol = protocol;
        return this;
    }

    public NetworkPort setIface(NetworkInterfaceType iface) {
        this.iface = iface;
        return this;
    }
}

File: src/main/java/org/cobbzilla/util/network/NetworkPort.kt
package org.cobbzilla.util.network

import lombok.experimental.Accessors

@Accessors(chain = true)
class NetworkPort {

    internal var port: Int? = null
    internal var protocol = TransportProtocol.tcp
    internal var iface = NetworkInterfaceType.world

    fun getPort(): Int? {
        return this.port
    }

    fun getProtocol(): TransportProtocol {
        return this.protocol
    }

    fun getIface(): NetworkInterfaceType {
        return this.iface
    }

    fun setPort(port: Int?): NetworkPort {
        this.port = port
        return this
    }

    fun setProtocol(protocol: TransportProtocol): NetworkPort {
        this.protocol = protocol
        return this
    }

    fun setIface(iface: NetworkInterfaceType): NetworkPort {
        this.iface = iface
        return this
    }
}
